corehq/form_processor/backends/sql/ledger.py

```diff
@@ -77,6 +77,7 @@ def _lazy_original_balance():
         ledger_value.track_create(transaction)
         # only do this after we've created the transaction otherwise we'll get the wrong delta
         ledger_value.balance = new_ledger_values.balance
+        ledger_value.last_modified = stock_trans.timestamp
```

Comment:
Good question. `stock_trans.timestamp` is actually the date from the stock report entry in the XML:
```xml
<balance date="date from form" entity-id="..." section-id="...">
	....
</balance>
```
This change copies the current functionality: https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/stock/signals.py#L113

I think the date is only used to show in reports and also in generating the restore XML: https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/phone/data_providers/case/stock.py#L42

To be honest I would prefer this to match up with `form.received_on` but I'm not sure if that would break anything on mobile (though I don't expect it would)